lib/pbench/agent/tool_meister.py

```diff
@@ -463,11 +512,19 @@ def _get_data(self):
             self.logger.debug("next")
             try:
                 payload = next(self._chan)
-            except Exception:
-                # FIXME: Add connection drop error handling, retry loop
-                # re-establishing a connection.
-                self.logger.exception("Error fetching 'next' data off channel")
-                continue
+            except redis.ConnectionError as exc:
+                try:
+                    msg = exc.args[0]
+                except Exception:
+                    msg = ""
+                if msg.startswith(SERVER_CLOSED_CONNECTION_ERROR):
+                    # Redis disconnected, shutdown as gracefully as possible.
+                    # FIXME: Consider adding connection drop error handling, retry loop
```

Comment:
We use `next(self._chan)` to give us a chance to try to handle re-tries in the future.